test/test_tensordictclass.py

```diff
@@ -67,6 +67,24 @@ def test_attributes():
     assert equality_tensordict.batch_size == torch.Size(batch_size)
 
 
+def test_indexing():
+    data = MyData(
+        X=torch.ones(3, 4, 5),
+        y=torch.zeros(3, 4, 5, dtype=torch.bool),
+        batch_size=[3, 4],
+    )
+
+    assert data[:2].batch_size == torch.Size([2, 4])
+    assert data[1].batch_size == torch.Size([4])
+    assert data[1][1].batch_size == torch.Size([])
+
+    with pytest.raises(Exception):
```

Comment:
let's be more explicit with the exception type and the message (e.g. using `pytest.raises(ValueError, match="etc")`)